File: src/main/java/com/procurement/clarification/model/dto/UpdateEnquiryDto.java
package com.procurement.clarification.model.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import lombok.Getter;

@Getter
public class UpdateEnquiryDto {

    @Valid
    @NotNull
    @JsonProperty("enquiry")
    private final EnquiryAnswerDto enquiry;

    @JsonCreator
    public UpdateEnquiryDto(@JsonProperty("enquiry") final EnquiryAnswerDto enquiry) {
        this.enquiry = enquiry;
    }
}

File: src/main/kotlin/com/procurement/clarification/model/dto/UpdateEnquiryDto.kt
package com.procurement.clarification.model.dto

import com.fasterxml.jackson.annotation.JsonCreator
import javax.validation.Valid
import javax.validation.constraints.NotNull

data class UpdateEnquiryDto @JsonCreator constructor(

        @field:Valid
        @field:NotNull
        val enquiry: EnquiryAnswerDto
)
